src/gc/env/gcenv.os.h

```diff
@@ -50,6 +50,14 @@ struct GCThreadAffinity
 // An event is a synchronization object whose state can be set and reset
 // indicating that an event has occured. It is used pervasively throughout
 // the GC.
+//
+// Note that GCEvent deliberately leaks its contents by not having a non-trivial destructor.
+// This is by design; since many uses of GCEvent have static lifetime, their destructors
```

Comment:
Right - today GCEvent is embedded in classes that may not necessarily have static lifetime, but do in practice because their object graph is rooted by something with static lifetime (this is the case for `gc_heap` with Server GC). I'll be more clear about that.